File: backend/src/main/java/com/example/pixelplay/chess/mechanics/BishopMechanics.java
package com.example.pixelplay.chess.mechanics;

import com.example.pixelplay.chess.Position;
import com.example.pixelplay.chess.PositionUtil;

import java.util.ArrayList;
import java.util.List;

public class BishopMechanics extends StraightMechanics implements PieceMechanics {
    public BishopMechanics(Position position, byte square) {
        super(position, square);
    }

    @Override
    protected List<MoveDirection> getMovingDirection() {
        return MoveDirection.diagonalMoves;
    }
}

File: backend/src/main/java/com/example/pixelplay/chess/mechanics/BishopMechanics.kt
package com.example.pixelplay.chess.mechanics

import com.example.pixelplay.chess.Position

class BishopMechanics() : StraightMechanics(), PieceMechanics {
    override val movingDirection: List<Cell>
        get() = Cell.diagonalMoves

    override fun moves(position: Position, index: Byte): List<Byte> {
        TODO("Not yet implemented")
    }
}
